### PR TITLE
Upload project binary with github release

### PR DESCRIPTION
**Pull request by sslavic**
This PR adds support for publishing project binary with github releases.
The binary will be downloaded and used by e2e tests. First planned use will be to replace minikube with e2ectl/kind in https://github.com/giantswarm/kubernetes-coredns/blob/master/.circleci/config.yml

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,3 +32,26 @@ jobs:
...
+  github-release:
+    working_directory: /home/circleci/.go_workspace/src/github.com/giantswarm/e2ectl
+    machine: true
+    steps:
+      - checkout
+
+      - attach_workspace:
+          at: .
+
+      - run: ./github-release.sh $CIRCLE_PROJECT_REPONAME $CIRCLE_SHA1 $PERSONAL_ACCESS_TOKEN
```
Comment (kopiczko) [inline .circleci/config.yml]: @TheoBrigitte WDYT about extracting architect command to do a release? We are doing it already for tagged commits.

**Review by kopiczko — APPROVED** [review]
Not blocking

File: .circleci/config.yml (hunk 1 above)
Comment (sslavic) [reply .circleci/config.yml]: I chose consistency with other GS Go projects.

Curious was it considered to use/switch to Go community defacto standard - https://goreleaser.com/ ?

Comment (kopiczko) [reply .circleci/config.yml]: > Curious was it considered to use/switch to Go community defacto standard - https://goreleaser.com/ ?

I don't think so. But I'd like to see us using it. Especially if this is encapsulated into an orb.

**Pull request merged**